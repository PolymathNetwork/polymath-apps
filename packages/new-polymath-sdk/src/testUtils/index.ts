import Web3PromiEvent from 'web3-core-promievent';
import { GenericContract } from '~/LowLevel/types';
import { PostTransactionResolver } from '~/PostTransactionResolver';

export class MockedContract<T extends GenericContract> {
  public autoResolve: boolean;
  public errorMsg?: string;
  public fakeTxOnePromiEvent = new Web3PromiEvent();
  public fakeTxTwoPromiEvent = new Web3PromiEvent();
  public failureTxPromiEvent = new Web3PromiEvent();

  public fakeTxOne = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxOnePromiEvent.resolve();
      }
      return this.fakeTxOnePromiEvent.eventEmitter;
    };
  });
  public fakeTxTwo = jest.fn(async () => {
    return () => {
      if (this.autoResolve) {
        this.fakeTxTwoPromiEvent.resolve();
      }
      return this.fakeTxTwoPromiEvent.eventEmitter;
    };
  });

  public failureTx = jest.fn(async () => {
    return () => {
      const err = this.errorMsg || 'Test error';
      this.failureTxPromiEvent.reject(new Error(err));
      return this.failureTxPromiEvent.eventEmitter;
    };
  });

  constructor({
    autoResolve = true,
    errorMsg,
  }: { autoResolve?: boolean; errorMsg?: string } = {}) {
    this.autoResolve = autoResolve;
    this.errorMsg = errorMsg;
  }
}

export const getMockTransactionSpec = (
  method: (args: any) => Promise<any>,
  args: any,
  resolver = async () => {}
) => ({
  method,
  args,
  postTransactionResolver: new PostTransactionResolver(resolver),
});
