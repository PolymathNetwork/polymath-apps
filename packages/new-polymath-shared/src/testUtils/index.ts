export * from './factories';

const originalWindow = {
  ...window,
};

interface WindowWithEthereum extends Window {
  ethereum: {
    networkVersion: string;
    _metamask?: {
      isApproved: () => Promise<boolean>;
    };
    enable(): Promise<any>;
  };
}

interface WindowWithWeb3 extends Window {
  web3: {
    version: string;
  };
}

interface MockEthereumBrowserArgs {
  support?: 'legacy' | 'modern';
  options?: {
    networkId: number;
    loaded: boolean;
  };
}

export function mockEthereumBrowser({
  support = 'modern',
  options = {
    networkId: 1,
    loaded: true,
  },
}: MockEthereumBrowserArgs) {
  const { networkId, loaded } = options;

  if (support === 'modern') {
    window = (Object.assign(window, {
      ethereum: {
        networkVersion: loaded ? `${networkId}` : undefined,
      },
    }) as any) as WindowWithEthereum;
  }
  if (support === 'legacy') {
    window = Object.assign(window, {
      web3: {
        version: networkId,
      },
    } as any) as WindowWithWeb3;
  }

  return {
    restore: () => {
      window = originalWindow;
    },
    load: () => {
      if (support === 'modern') {
        (window as WindowWithEthereum).ethereum.networkVersion = `${networkId}`;
      }
      if (support === 'legacy') {
        (window as WindowWithWeb3).web3.version = `${networkId}`;
      }
    },
  };
}
