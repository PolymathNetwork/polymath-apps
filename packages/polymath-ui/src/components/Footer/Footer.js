// @flow

import React from 'react';
import styled, { css } from 'styled-components';

import PageWrap from '../PageWrap';
import Flex from '../Flex';

import { textLinkInverted } from '../../styles/utils';

type Props = {|
  title?: string,
  termsOfService?: string,
  privacyPolicy?: string,
  variant?: string,
|};

const Container = styled.div`
  width: 100%;
  height: ${({ theme }) => theme.footer.height};
  ${({ variant }) => variants[variant].Container};

  .pui-footer-text {
    display: inline-flex;
    font-size: 14px;
    line-height: 22px;
  }
`;

const Inner = styled(Flex)`
  height: ${({ theme }) => theme.footer.height};
`;

const Links = styled.ul`
  margin-left: auto;
  display: inline-flex;
  ${({ variant }) => variants[variant].Links};

  li {
    display: inline-flex;
    align-items: center;
    font-size: 14px;
    line-height: 22px;
    margin-left: 30px;

    a {
      ${textLinkInverted};
    }
  }
`;

const variants = {
  default: {
    Container: css`
      background-color: white;
      box-shadow: 0 -1px 0 0 #dfe3e6;
    `,
    Links: css`
      a {
        color: ${({ theme }) => theme.colors.baseText};
      }
    `,
  },
  transparent: {
    Links: css`
      a {
        color: white;
      }
    `,
  },
};

const Footer = (props: Props) => {
  return (
    <Container className="pui-footer" variant={props.variant}>
      <PageWrap>
        <Inner>
          <span className="pui-footer-text">
            &copy; {new Date().getFullYear()} {props.title || 'Polymath'}
          </span>
          <Links className="pui-footer-links" variant={props.variant}>
            <li>
              <a
                href={
                  props.termsOfService ||
                  'https://polymath.network/termsofservice.html'
                }
                target="_blank"
                rel="noopener noreferrer"
              >
                Terms Of Service
              </a>
            </li>
            <li>
              <a
                href={
                  props.privacyPolicy ||
                  'https://polymath.network/privacypolicy.html'
                }
                target="_blank"
                rel="noopener noreferrer"
              >
                Privacy Policy
              </a>
            </li>
          </Links>
        </Inner>
      </PageWrap>
    </Container>
  );
};

export default Footer;

Footer.defaultProps = {
  variant: 'default',
};
