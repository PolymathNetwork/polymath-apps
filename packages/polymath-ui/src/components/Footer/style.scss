@import '~@polymathnetwork/ui/src/styles/_variables.scss';

.pui-footer {
  z-index: 10;
  width: 100%;
  height: $navbar-height;
  background-color: white;
  border-top: 1px solid #E1E5E8;
  position: fixed;
  bottom: 0;
  left: 0;

  .pui-footer-links {
    float: right;
    margin-right: 56px;
    margin-left: 219px;
    display: inline-flex;
    margin-top: 11px;

    li {
      display: inline-flex;
      align-items: center;
      color: $poly-gray-dark;
      font-size: 14px;
      line-height: 22px;
      margin-left: 30px;
      font-weight: 300;

      a {
        text-decoration: none;
        color: $poly-gray-dark;
      }
    }
  }

  .pui-footer-text {
    display: inline-flex;
    color: $poly-gray-dark;
    font-size: 14px;
    line-height: 22px;
    font-weight: 300;
    margin-left: 120px;
    margin-top: 11px;
  }
}