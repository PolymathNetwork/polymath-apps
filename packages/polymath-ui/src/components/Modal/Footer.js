import React from 'react';

import Box from '../Box';

const Footer = props => {
  return <Box textAlign="right">{props.children}</Box>;
};

export default Footer;
