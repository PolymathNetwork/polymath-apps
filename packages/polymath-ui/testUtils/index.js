import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'styled-components';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import theme from '../src/theme';

const middlewares = [thunk];

const customRender = (node, ...opts) => {
  const { rerender, ...result } = render(
    <ThemeProvider theme={theme}>{node}</ThemeProvider>,
    ...opts
  );

  const customRerender = (rerenderNode, ...rerenderOpts) => {
    return rerender(
      <ThemeProvider theme={theme}>{rerenderNode}</ThemeProvider>,
      ...rerenderOpts
    );
  };

  return { rerender: customRerender, ...result };
};

export const mockStore = configureMockStore(middlewares);
export * from 'react-testing-library';
export { customRender as render };
