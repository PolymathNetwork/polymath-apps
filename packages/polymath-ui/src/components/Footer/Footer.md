---
name: Footer
---

import { Playground, PropsTable } from 'docz';
import Footer from './';

# Footer

<PropsTable of={Footer} />

## Basic usage

<Playground>
  <style>{`.pui-footer { position: static; }`}</style>
  <Footer />
</Playground>
