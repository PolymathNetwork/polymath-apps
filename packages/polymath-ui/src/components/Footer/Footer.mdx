---
name: Footer
---

import { Playground, PropsTable } from 'docz';
import Footer from './';

# Footer

<PropsTable of={Footer} />

## Basic usage

<style>{`.pui-footer { position: static; }`}</style>

<Playground>
  <Footer />
</Playground>
