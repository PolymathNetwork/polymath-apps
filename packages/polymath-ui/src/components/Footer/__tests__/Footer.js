import React from 'react';
import { MemoryRouter } from 'react-router';

import { render } from '../../../../testUtils';

import Footer from '../Footer';

test('renders without crashing', () => {
  const { container } = render(
    <MemoryRouter>
      <Footer />
    </MemoryRouter>
  );

  expect(container.firstChild).toMatchSnapshot();
});
