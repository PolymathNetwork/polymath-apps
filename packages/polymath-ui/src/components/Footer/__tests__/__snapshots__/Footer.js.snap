// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="pui-footer Footer__Container-hx8agb-0 jKXVoU"
>
  <div
    class="Box-sc-1ayjt33-0 PageWrap-j7u0ds-0 koPitZ"
  >
    <div
      class="Box-sc-1ayjt33-0 Flex-sc-1kg4q90-0 Footer__Inner-hx8agb-1 giiVdw"
    >
      <span
        class="pui-footer-text"
      >
        Â© 
        2019
         
        Polymath
      </span>
      <ul
        class="pui-footer-links Footer__Links-hx8agb-2 fjOXNy"
      >
        <li>
          <a
            href="https://polymath.network/termsofservice.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Terms Of Service
          </a>
        </li>
        <li>
          <a
            href="https://polymath.network/privacypolicy.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Privacy Policy
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
